week05/my-todolist/src/App.js

```diff
@@ -3,21 +3,65 @@ import "./App.css";
 import Header from "./Header";
 import TodoList from "./TodoList";
 import TodoAdd from "./TodoAdd";
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 
 const TODO_LIST_KEY = "todoList";
 
+// 커스텀 훅 사용하지 않는 코드
+/*const [theme, setTheme] = useState("light");
+  const [buttonEmoji, setButtonEmoji] = useState("🌞");
+
+  const toggleTheme = useCallback(() => {
+    if (theme == "light") {
+      setTheme("dark");
+      setButtonEmoji("🌛");
+    } else if (theme == "dark") {
+      setTheme("light");
+      setButtonEmoji("🌞");
+    }
+  }, [theme]);
+  useEffect(() => {
+    // theme 상태가 변경될 때마다 배경색을 변경
+    document.documentElement.style.backgroundColor =
+      theme === "light" ? "#f2eae4" : "black";
+  }, [theme]);
+  useEffect(() => {
+    localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
+  }, [todoList]);*/
+
+// useTheme이라는 이름의 커스텀 훅 생성
+function useTheme(theme, toggleTheme) {
+  useEffect(() => {
+    document.documentElement.style.backgroundColor =
+      theme === "light" ? "#f2eae4" : "black";
+  }, [theme, toggleTheme]);
+}
+
 function App() {
   const [todoList, setTodoList] = useState(() => {
     const saveTodoList = localStorage.getItem(TODO_LIST_KEY);
     return saveTodoList ? JSON.parse(saveTodoList) : [];
   });
 
-  useEffect(() => {
-    localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
-  }, [todoList]);
+  const [theme, setTheme] = useState("light");
+  const [buttonEmoji, setButtonEmoji] = useState("🌞");
+
+  const toggleTheme = useCallback(() => {
+    if (theme == "light") {
+      setTheme("dark");
+      setButtonEmoji("🌛");
+    } else if (theme == "dark") {
+      setTheme("light");
+      setButtonEmoji("🌞");
+    }
+  }, [theme]);
+
+  useTheme(theme, toggleTheme);
```

Comment:
```suggestion
  const { toggleTheme, buttonEmoji } = useTheme();
```
커스텀 훅의 사용 의의는 기능별로 파일을 분리하여 보다 깔끔하게 코드를 유지하기 위함에 있습니다. 따라서 테마와 관련된 모든 코드는 useTheme 커스텀 훅 파일로 옮기고, 이 커스텀 훅에서 관리 중인 상태나 함수 중 App.js와 같이 사용하는 파일에서 필요한 값만 가져와서 사용해야 합니다.